### PR TITLE
Empty Feature Grid

### PR DESCRIPTION
**Pull request by NicoRenaud**
Changed the DataGenerator and GridTools to skip empty features. I also removed the non xyz format in GridTools since I think we only use xyz now

### CODE REVIEW

**Comment by NicoRenaud** [comment]
@LilySnow before we merge could you check if that works on the pdb you sent me ? I did work locally on my laptop but it would be good to run extended check before merging

File: deeprank/generate/GridTools.py [hunk 1]
```diff
@@ -554,7 +546,7 @@ def map_features(self, featlist, transform=None):
                     dict_data[feature_name + '_chainB'] = np.zeros(self.npts)
                 else:
                     dict_data[feature_name] = np.zeros(self.npts)
-            else:
+            else: # do we need that ?!
```
Comment (CunliangGeng) [inline deeprank/generate/GridTools.py:549]: It's never used before and we never tried `transform` neither. It's fine to keep it.

**Comment by CunliangGeng** [comment]
The code skips the feature if it does not have value, which means the code can generate HDF5 file but `features` and `mapped_features` will not be uniformed (same size) for different docking decoys. This will cause problem for the learning step.

**Comment by NicoRenaud** [comment]
Well it does create an empty grid for features where there are not atoms no ?

**Comment by CunliangGeng** [comment]
> Well it does create an empty grid for features where there are not atoms no ?

My fault, you'r right.  Those mapped features will be zero.

**Pull request merged**